### PR TITLE
[Buttons] Respect minimum touch target size

### LINKED ISSUE #223: Buttons sizeToFit/sizeThatFits should enforce minimum touch sizes
Buttons should respect the minimum touch size of Material: https://www.google.com/design/spec/layout/metrics-keylines.html#metrics-keylines-touch-target-size

### PR DESCRIPTION
**Pull request by romoore**
All Buttons should support a minimum touch target size of 48 points for
both width and height.  Unless custom |hitAreaInsets| are set on the
button, the touch target will be expanded to 48x48 points and centered
around the button's center.

Closes #223

### CODE REVIEW

**Comment by romoore** [comment]
Marking as "author will merge" because this is a significant proposed change in behavior, particularly for "mini" Floating Action Buttons.

File: components/Buttons/tests/unit/ButtonsTests.m [hunk 1]
```diff
@@ -276,15 +275,14 @@ - (void)testCurrentElevationHighlighted {
   CGFloat normalElevation = 10;
   CGFloat elevation = 40;
   [button setElevation:normalElevation forState:UIControlStateNormal];
-  [button setElevation:elevation forState:UIControlStateHighlighted];;
+  [button setElevation:elevation forState:UIControlStateHighlighted];
+  ;
```
Comment (samnm) [inline components/Buttons/tests/unit/ButtonsTests.m]: ;

There are two more below also.

**Review by samnm — APPROVED** [review]
Some tiny cleanup changes requested, otherwise LGTM.

File: components/Buttons/tests/unit/ButtonsTests.m (hunk 1 above)
Comment (romoore) [reply components/Buttons/tests/unit/ButtonsTests.m]: Done

**Review by ianegordon — CHANGES_REQUESTED** [review]
Would you please add a test and/or example testing what happens when buttons smaller than the minimum size are adjacent? 
I immediately thought of a keypad-like interface with a 3 x 3 grid of number buttons claiming touches intended for their neighbor.

**Comment by romoore** [comment]
This approach is too surprising. I will push up another CR that takes a gentler approach.
